contracts/chain-adapters/utilities/HubPoolStore.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { HubPoolInterface } from "../../interfaces/HubPoolInterface.sol";
+
+interface IHubPool {
+    function rootBundleProposal() external view returns (HubPoolInterface.RootBundle memory);
+}
+
+/**
+ * @notice Stores data that can be relayed to L2 SpokePool using storage proof verification and light client contracts
+ * on the L2 where the SpokePool is deployed. Only the HubPool can store data to this contract.
+ * @dev Designed to be used with UniversalStorageProof_Adapter and UniversalStorageProof_SpokePool.
+ * @dev This contract DOES NOT prevent replay attacks of storage proofs on the L2 spoke pool if the
+ * UniversalStorageProof_Adapters using this contract are mapped to spokepools with the same address on different
+ * L2 chains. See comment in storeRelayAdminFunctionCalldata() for more details.
+ */
+contract HubPoolStore {
+    error NotHubPool();
+
+    // Maps unique calldata hashes to calldata.
+    mapping(bytes32 => bytes) public relayAdminFunctionCalldata;
```

Comment:
I think this is inconsistent with the `SpokePool` verification logic, right?

In the SP, we assume a hash is stored in this slot, right, but here we store the full bytes value, which would require us to verify multiple slots, right?

Reply:
oh true, in the SP we should assume the slot is stored as a `bytes` not a `bytes32` i'll correct that

Reply:
6f3f9177c0cc3337cd4ac3dc563a5f8746397ca4